### PR TITLE
Update RAPIDS to 25.02.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR updates RAPIDS to 25.02 and updates the RAPIDS dependency tree used for CI testing.

## Checklist
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
@trxcllnt Can you help me figure out why this is failing? I think `cugraph-gnn` is failing to clone because of dashes in the repo name. I looked through this repo and the `rapidsai/devcontainers` repo but I am struggling to trace the bash code all the way through to find the point of failure.

**Comment by bdice** [comment]
https://github.com/rapidsai/devcontainers/pull/427 should fix the issues I am seeing.

File: ci/rapids/cuda12.5-conda/devcontainer.json [hunk 1]
```diff
@@ -36,12 +36,11 @@
...
     "RAPIDS_cumlprims_mg_GIT_REPO": "${localEnv:RAPIDS_cumlprims_mg_GIT_REPO}",
     "RAPIDS_cuml_GIT_REPO": "${localEnv:RAPIDS_cuml_GIT_REPO}",
-    "RAPIDS_cugraph_ops_GIT_REPO": "${localEnv:RAPIDS_cugraph_ops_GIT_REPO}",
-    "RAPIDS_wholegraph_GIT_REPO": "${localEnv:RAPIDS_wholegraph_GIT_REPO}",
     "RAPIDS_cugraph_GIT_REPO": "${localEnv:RAPIDS_cugraph_GIT_REPO}",
+    "RAPIDS_cugraph_gnn_GIT_REPO": "${localEnv:RAPIDS_cugraph_gnn_GIT_REPO}",
     "RAPIDS_cuspatial_GIT_REPO": "${localEnv:RAPIDS_cuspatial_GIT_REPO}"
   },
-  "initializeCommand": ["/bin/bash", "-c", "mkdir -m 0755 -p ${localWorkspaceFolder}/.{aws,cache,config} ${localWorkspaceFolder}/ci/rapids/.{conda,log/devcontainer-utils} ${localWorkspaceFolder}/ci/rapids/.repos/{rmm,kvikio,ucxx,cudf,raft,cuvs,cuml,wholegraph,cugraph,cuspatial}"],
+  "initializeCommand": ["/bin/bash", "-c", "mkdir -m 0755 -p ${localWorkspaceFolder}/.{aws,cache,config} ${localWorkspaceFolder}/ci/rapids/.{conda,log/devcontainer-utils} ${localWorkspaceFolder}/ci/rapids/.repos/{rmm,kvikio,ucxx,cudf,raft,cuvs,cuml,cugraph,cugraph-gnn,cuspatial}"],
```
Comment (trxcllnt) [inline ci/rapids/cuda12.5-conda/devcontainer.json:43]:
```suggestion
  "initializeCommand": ["/bin/bash", "-c", "mkdir -m 0755 -p ${localWorkspaceFolder}/.{aws,cache,config} ${localWorkspaceFolder}/ci/rapids/.{conda,log/devcontainer-utils} ${localWorkspaceFolder}/ci/rapids/.repos/{rmm,kvikio,ucxx,cudf,raft,cuvs,cuml,cugraph,cugraph-gnn,nx-cugraph,cuspatial}"],
```

File: ci/rapids/cuda12.5-conda/devcontainer.json [hunk 2]
```diff
@@ -57,8 +56,8 @@
     "source=${localWorkspaceFolder}/ci/rapids/.repos/raft,target=/home/coder/raft,type=bind,consistency=consistent",
     "source=${localWorkspaceFolder}/ci/rapids/.repos/cuvs,target=/home/coder/cuvs,type=bind,consistency=consistent",
     "source=${localWorkspaceFolder}/ci/rapids/.repos/cuml,target=/home/coder/cuml,type=bind,consistency=consistent",
-    "source=${localWorkspaceFolder}/ci/rapids/.repos/wholegraph,target=/home/coder/wholegraph,type=bind,consistency=consistent",
     "source=${localWorkspaceFolder}/ci/rapids/.repos/cugraph,target=/home/coder/cugraph,type=bind,consistency=consistent",
+    "source=${localWorkspaceFolder}/ci/rapids/.repos/cugraph-gnn,target=/home/coder/cugraph-gnn,type=bind,consistency=consistent",
```
Comment (trxcllnt) [inline ci/rapids/cuda12.5-conda/devcontainer.json:60]:
```suggestion
    "source=${localWorkspaceFolder}/ci/rapids/.repos/cugraph-gnn,target=/home/coder/cugraph-gnn,type=bind,consistency=consistent",
    "source=${localWorkspaceFolder}/ci/rapids/.repos/nx-cugraph,target=/home/coder/nx-cugraph,type=bind,consistency=consistent",
```

File: ci/rapids/cuda12.5-conda/devcontainer.json [hunk 3]
```diff
@@ -36,12 +36,11 @@
     "RAPIDS_cuvs_GIT_REPO": "${localEnv:RAPIDS_cuvs_GIT_REPO}",
     "RAPIDS_cumlprims_mg_GIT_REPO": "${localEnv:RAPIDS_cumlprims_mg_GIT_REPO}",
     "RAPIDS_cuml_GIT_REPO": "${localEnv:RAPIDS_cuml_GIT_REPO}",
-    "RAPIDS_cugraph_ops_GIT_REPO": "${localEnv:RAPIDS_cugraph_ops_GIT_REPO}",
-    "RAPIDS_wholegraph_GIT_REPO": "${localEnv:RAPIDS_wholegraph_GIT_REPO}",
     "RAPIDS_cugraph_GIT_REPO": "${localEnv:RAPIDS_cugraph_GIT_REPO}",
+    "RAPIDS_cugraph_gnn_GIT_REPO": "${localEnv:RAPIDS_cugraph_gnn_GIT_REPO}",
```
Comment (trxcllnt) [inline ci/rapids/cuda12.5-conda/devcontainer.json:40]:
```suggestion
    "RAPIDS_cugraph_gnn_GIT_REPO": "${localEnv:RAPIDS_cugraph_gnn_GIT_REPO}",
    "RAPIDS_nx_cugraph_GIT_REPO": "${localEnv:RAPIDS_nx_cugraph_GIT_REPO}",
```

File: ci/rapids/cuda12.5-conda/devcontainer.json (hunk 1 above)
Comment (bdice) [reply ci/rapids/cuda12.5-conda/devcontainer.json:43]: I was going to skip nx-cugraph since it’s only a Python package. Does it use CCCL?

Comment (trxcllnt) [reply ci/rapids/cuda12.5-conda/devcontainer.json:44]: Good point, it looks like we only build the C++ libraries.

**Comment by miscco** [comment]
>Error:  Dev Container files are out of date or there are untracked files. Run the .devcontainer/make_devcontainers.sh script and commit the changes.

**Comment by bernhardmgruber** [comment]
Since the CI is green, is this ready for review?

**Comment by bdice** [comment]
@bernhardmgruber Yes, this is now ready for review. Sorry, just juggling a lot of PRs and didn't want to waste anyone's time before it was all ready. 👍

**Review by bernhardmgruber — APPROVED** [review]

**Comment by bernhardmgruber** [comment]
Let's have a second opinion, since I am not so versed in CI and devcontainers.

**Review by miscco — APPROVED** [review]

**Pull request merged**